br/pkg/lightning/backend/local/local.go

```diff
@@ -1342,9 +1342,19 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID, regi
 		log.L().Info("engine contains no kv, skip import", zap.Stringer("engine", engineUUID))
 		return nil
 	}
-	regionSplitKeys := int64(regionMaxKeyCount)
-	if regionSplitSize > defaultRegionSplitSize {
-		regionSplitKeys = int64(float64(regionSplitSize) / float64(defaultRegionSplitSize) * float64(regionMaxKeyCount))
+	kvRegionSplitSize, kvRegionSplitKeys, err := getRegionSplitSizeKeys(ctx, local.pdCtl.GetPDClient(), local.tls)
+	if err == nil {
+		if kvRegionSplitSize > regionSplitSize {
+			regionSplitSize = kvRegionSplitSize
+		}
+		if kvRegionSplitKeys > regionSplitKeys {
+			regionSplitKeys = kvRegionSplitKeys
+		}
+	} else {
+		log.L().Warn("fail to get region split keys and size", zap.Error(err))
+		if regionSplitSize > defaultRegionSplitSize {
+			regionSplitKeys = int64(float64(regionSplitSize) / float64(defaultRegionSplitSize) * float64(regionMaxKeyCount))
+		}
```

Comment:
It seems `local.defaultRegionSplitSize` and `local.regionMaxKeyCount` are the same with `config.SplitRegionSize` and `config.SplitRegionKeys`. 

Can we just remove  `local.defaultRegionSplitSize` and `local.regionMaxKeyCount` ?

Reply:
addressed in https://github.com/pingcap/tidb/pull/34537/commits/0faf095984245ac18e70633993537b21315b5313